src/core/derivation.ts

```diff
@@ -181,9 +181,21 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T, con
     }
     globalState.trackingDerivation = prevTracking
     bindDependencies(derivation)
+
+    if (derivation.observing.length === 0) {
+        warnAboutDerivationWithoutDependencies(derivation.name)
+    }
+
     return result
 }
 
+function warnAboutDerivationWithoutDependencies(name: string) {
+    if (process.env.NODE_ENV === "production") return
+    if (globalState.derivationRequiresObservable) {
+        console.warn(`[mobx] Derivation ${name} is created without reading any observable value`)
```

Comment:
I wonder if we can provide information given that name is usually generated and doesn't make much sense. It would make sense to spill out stack trace at least. Same for the other warning.

Reply:
I took the behaviour of `computedRequiresReaction`,
But i don't have any objection to add it across the board.
We can add to the docs that you can use the debugger to add a breakpoint to see the call stack

Reply:
> We can add to the docs that you can use the debugger to add a breakpoint to see the call stack

Um, that's hardly a good DX :) I believe the majority of people still don't know how to use debugger properly.

Alright, let's keep it like that in this PR and improve later. I am especially curious how well it will integrate with `mobx-react` and usefulness of these warnings.